tvtk/wrapper_gen.py

```diff
@@ -511,223 +691,178 @@ def _gen_get_set_methods(self, klass, out):
-                if sig[0][1] is None:
-                    ret = sig[0][0][0]
-                    if ret[:3] == 'vtk':
-                        default, rng = None, None
-                    elif ret == 'int':
-                        default, rng = 0, None
-                    elif ret == 'float':
-                        default, rng = 0.0, None
-                    elif ret == 'string':
-                        default, rng = '', None
-                    else:
-                        default, rng = None, None
+                # In some rare cases, the vtk class is an abstract class
+                # that does not have a concrete subclass
+                # We patch the default using the get/set method
+                # while the below seems a hack, this is the best we could do
+                default, rng = (patch_default(vtk_get_meth, vtk_set_meth, None),
+                                None)
+
+            # --------------------------------------------------------
+            # Has a specified range of valid values.  Write and done
+            # --------------------------------------------------------
+            if rng:
+                self._write_trait_with_range(klass, out, vtk_attr_name)
+                continue
+
+            # ----------------------------------------------------------
+            # The VTK Get method for the attribute returns the address
+            # to a pointer, this is therefore, a VTK python bug
+            # ----------------------------------------------------------
+            if isinstance(default, str) and default.endswith('_p_void'):
+                try:
+                    self._write_trait_with_default_undefined(klass, out,
+                                                             vtk_attr_name)
+                except TypeError as exception:
+                    # We could not write the trait
+                    # we will let the next clause handle it
+                    print('Warning:', str(exception))
+                    default = None
                 else:
-                    self._write_tvtk_method(klass, out, vtk_get_meth, sig)
-                    self._write_tvtk_method(klass, out, vtk_set_meth)
+                    # Getting the trait value using the get method
+                    # without argument will continue to return an address to
+                    # a pointer, so there is no point in updating.
+                    # However, some VTK Get methods have multiple signatures
+                    # among which you could pass a numpy array of the right
+                    # size and the Get method would populate the array with
+                    # the value of the attribute
+                    # e.g. x = numpy.empty(9); ImageConvolve.GetKernel3x3(x)
+                    # FIXME: we could try harder in retrieving these values
+                    # in TVTKBase.update_traits
+                    del updateable_traits[name]
```

Comment:
If I change `del updateable_traits[name]` to `allow_update_failure.add(name)`, the difference in behaviour is not only that TVTKBase.update_traits will have more updating to do while those updating will almost certainly fail with TraitError, but I will also get these traits visual on the UI.
E.g. with `del updateable_traits[name]`
![del_updateable](https://cloud.githubusercontent.com/assets/3673984/15048898/6f1c459c-12e3-11e6-8942-348c4d820594.png)

E.g. replace with `allow_update_failure.add(name)`
![allow](https://cloud.githubusercontent.com/assets/3673984/15048870/4de5ab34-12e3-11e6-9693-3b1e76a8bd0f.png)

Question is, which one do we want.  The bahaviour on the interactive interpreter is the same.